scalafmt-core/shared/src/main/scala/org/scalafmt/Scalafmt.scala

```diff
@@ -82,6 +92,15 @@ object Scalafmt {
     }
   }
 
+  // XXX: don't modify signature, scalafmt-dynamic expects it via reflection
+  def format(
```

Comment:
This looks like a new signature, did it exist in an older version of Scalafmt? Can you elaborate on how this fixes #1628

Reply:
@olafurpg per our standard "i screwed up" practice, i reverted the original commit that breaks the contract (that's commit 1) and then added a second one which mainly restores the functionality but without touching the signatures of the two .format methods, you can probably see it by reviewing via the commits tab rather than the changes tab.

@tanishiking posted a stack trace yesterday which shows the failure to be in trying to find .format with the three-parameter signature, via reflection: https://github.com/scalameta/scalafmt/issues/1628#issuecomment-579372180

While I, unfortunately, don't have a way to reproduce the problem, this is most likely the only reason for #1628.

Reply:
This regression had been introduced in https://github.com/scalameta/scalafmt/pull/1608 (sorry I missed this change), and this PR basically revert the root cause of this regression.

Basically, it's probably better to write a detailed description, so that even people (who don't follow the context) can understand the change :)

Reply:
Thanks for the explanation! I haven't followed all the recent changes closely. It's very important that the `Scalafmt.scala` file does not have binary breaking changes. The `scalafmt-dynamic` interface is an attempt to provide a richer public API that is binary stable, but build tools already interface with the `Scalafmt` object so we need to keep supporting it